cloud-yarn/src/main/java/com/sequenceiq/cloudbreak/cloud/yarn/ApplicationNameUtil.java

```diff
@@ -1,29 +1,61 @@
 package com.sequenceiq.cloudbreak.cloud.yarn;
 
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
+import java.util.List;
+
 @Component
 public class ApplicationNameUtil {
 
     @Value("${cb.max.yarn.resource.name.length:}")
     private int maxResourceNameLength;
 
+    public String createApplicationName(AuthenticatedContext authenticatedContext) {
+        List<String> applicationNameAndUser = initializeApplicationName(authenticatedContext.getCloudContext());
+        return decorateName(applicationNameAndUser.get(0), applicationNameAndUser.get(1));
+    }
+
+    public String createLoadBalancerName(AuthenticatedContext authenticatedContext) {
+        List<String> applicationNameAndUser = initializeApplicationName(authenticatedContext.getCloudContext());
+        return decorateLoadBalancerName(applicationNameAndUser.get(0), applicationNameAndUser.get(1));
+    }
+
+    private List<String> initializeApplicationName(CloudContext cloudContext) {
+        String name = cloudContext.getName();
+        String id = cloudContext.getId().toString();
+        name += "-" + id;
+        String user = cloudContext.getUserName().split("@")[0].replaceAll("[^a-z0-9-_]", "");
+        return List.of(name, user);
+    }
+
     public String decorateName(String name, String userName) {
         if (name.endsWith("-cb")) {
             name += "-" + userName;
         }
         if (!name.endsWith("-cb-" + userName)) {
             name = name + "-cb-" + userName;
         }
-        return cutToMaxLength(name, userName);
+        return cutToMaxLength(name, userName, "-cb-");
+    }
+
+    public String decorateLoadBalancerName(String name, String userName) {
+        if (name.endsWith("-lb")) {
+            name += "-" + userName;
```

Comment:
**nit:** Use assignment and plus rather than the `+=` operator.
```suggestion
            name = name + "-" + userName;
```

Reply:
Will do!